### PR TITLE
Fetch Data and Add Quotes

### PR DESCRIPTION
**Pull request by martinkarugaba**
# :factory_worker:  What I have worked on
- [ ] Fetched data from API
- [ ] Added quote to page
- [ ] Added loading and Error states
- [ ] Cleared all linter errors

### CODE REVIEW

File: src/context/reducer.js [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+        loading: false,
+        error: null,
+      };
+    case ACTIONS.FETCH_ERROR:
+      return {
+        ...state,
+        data: null,
+        loading: false,
+        error: true,
+      };
```
Comment (VICTOR-CHUKWU) [inline src/context/reducer.js:24]: - [ ] you actually did extra and it shows a good understanding of global state management :+1:

File: src/context/context.jsx [hunk 2]
```diff
@@ -1,20 +1,51 @@
...
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Api-Key': apiKey,
+        },
+      });
+      const data = await response.json();
+      dispatch({ type: ACTIONS.FETCH_SUCCESS, payload: data[0] });
+    } catch (error) {
+      dispatch({ type: ACTIONS.FETCH_ERROR, payload: error.message });
+    }
```
Comment (VICTOR-CHUKWU) [inline src/context/context.jsx:42]: - [ ] good job :clap:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @martinkarugaba , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] managed state with context API and usereducer

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/context/reducer.js (hunk 1 above)
Comment (martinkarugaba) [reply src/context/reducer.js:24]: - [ ] :pray:  Thank you @VICTOR-CHUKWU

File: src/context/context.jsx (hunk 2 above)
Comment (martinkarugaba) [reply src/context/context.jsx:42]: - [ ] :pray: Thanks @VICTOR-CHUKWU

**Pull request merged**